src/main/java/at/hannibal2/skyhanni/mixins/hooks/MouseSensitivityHook.kt

```diff
@@ -0,0 +1,72 @@
+package at.hannibal2.skyhanni.mixins.hooks
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.api.event.HandleEvent
+import at.hannibal2.skyhanni.events.DebugDataCollectEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+
+@SkyHanniModule
+object MouseSensitivityHook {
+    private val config get() = SkyHanniMod.feature.garden.sensitivityReducer
+    var state: MouseSensitivityState = MouseSensitivityState.DEFAULT
+        private set
+
+    private var lastIn: Float = Float.NaN
+    private var lastOut: Float = Float.NaN
+
+    private fun getMouseSensitivity(original: Float): Float {
+        if (original != lastIn) {
+            lastIn = original
+            lastOut = state.apply(original)
+        }
+
+        return lastOut
+    }
+
+    fun getMouseSensitivityWithWeirdMath(original: Float): Float {
+        val actualSensitivity = (original - 0.2f) / 0.6f
+
+        return getMouseSensitivity(actualSensitivity) * 0.6f + 0.2f
+    }
+
+    fun setState(newState: MouseSensitivityState) {
+        state = newState
+
+        lastIn = Float.NaN
+        lastOut = Float.NaN
+    }
+
+    @HandleEvent
+    fun onDebug(event: DebugDataCollectEvent) {
+        event.title("Mouse Sensitivity")
+
+        if (state == MouseSensitivityState.DEFAULT) {
+            event.addIrrelevant("not enabled")
+            return
+        }
+
+        event.addData {
+            add("current state: $state")
+        }
+    }
+
+    enum class MouseSensitivityState(
```

Comment:
the name "mouse" can go as this is already inside a class with the name mouse sensitivity